macros/Read_building_permits.sas

```diff
@@ -0,0 +1,135 @@
+		   lastmodifi=LastModified maraddress=MARAddress owner_name=OwnerName permit_app=ApplicantName permit_cat=PermitCategory
+		   permit_id=PermitID permit_sub=PermitSubcategory permit_typ=PermitType;
+run;
+
+proc freq data = permits_&year._in;
+	tables PermitType;
+run;
+
+/* Read in Census block shapefile */
+proc mapimport out=block_shp_in
+  datafile="&_dcdata_r_path\OCTO\Maps\Census_Blocks__2010.shp";  
+run;
+
+proc sort data=block_shp_in out=block_shp (keep = x y geoid10); 
+	by geoid10;
+run;
+
+
+/* Spatial join permits to Census blocks */
+proc ginside includeborder
+  data=permits_&year._in 
+  map=block_shp
+  out=permits_&year._join;
+  id geoid10;
+run;
+
+
+/* Add standard geographies to each permit, create dummy vars and label */
+data permits_&year._geo;
+	set permits_&year._join;
+
+	geoblk2010 = geoid10;
+
+	%Block10_to_anc02;
+	%Block10_to_anc12;
+	%Block10_to_bpk;
+	%Block10_to_city;
+	%Block10_to_cluster00;
+	%Block10_to_cluster_tr00;
+	%Block10_to_cluster17;
+	%Block10_to_eor;
+	%Block10_to_psa04;
+	%Block10_to_psa12;
+	%Block10_to_stantoncommons;
+	%Block10_to_tr00;
+	%Block10_to_tr10;
+	%Block10_to_vp12;
+	%Block10_to_ward02;
+	%Block10_to_ward12;
+	%Block10_to_zip;
+
+	label x = "Latitude (decimal degrees)"
+		  y = "Longitude (decimal degrees)"
+		  geoblk2010 = "Census block (2010)"
+		  appstatus = "Permit application status"
+		  descofwork = "Description of work"
+		  fees = "Amount of permit fees ($)"
+		  feetype = "Description and amount of fees required for permit"
+		  address = "Address on permit application"
+		  issuedate = "Permit issue date"
+		  lastmodified = "Last modified date and time"
+		  maraddress = "MAR address ID"
+		  objectID = "GIS ObjectID"
+		  Ownername = "Name of property owner"
+		  applicantname = "Name of permit applicant"
+		  permitcategory = "Permit category"
+		  permitid = "DCRA permit ID"
+		  permitsubcategory = "Permit subcategory"
+		  permittype = "Permit type"
+		  ssl = "SSL (square, suffix, lot)"
+		  zoning = "DC zoning code"
+		  xcoord = "Latitude (MD State Plane)"
+		  ycoord = "Longitude (MD State Plane)"
+	;
+
+	permit_&year.=1;
+
+	if PermitType= "CONSTRUCTION" then permit_construction_&year. = 1 ;
+	if PermitType= "HOME OCCUPATION" then permits_homeoccupation_&year.= 1 ;
+	if PermitType= "POST CARD" then permits_postcard_&year.= 1 ;
+	if PermitType= "SHOP DRAWING" then permits_shopdrawing_&year.= 1 ;
+	if PermitType= "SUPPLEMENTAL" then permits_supplemental_&year.= 1 ;
+
+	label permit_&year. = "Total building permits in &year."
+		  permit_construction_&year. = "Construction permits in &year."
+		  permits_homeoccupation_&year. = "Home occupation permits in &year."
+		  permits_postcard_&year. = "Postcard permits in &year."
+		  permits_shopdrawing_&year. = "Permit drawings in &year."
+		  permits_supplemental_&year. = "Supplemental building permits in &year."
+	;
```

Comment:
Ideally, should define some formats for the categorical variables *appstatus, permittype, permitcategory,* and *permitsubcategory* so that those values appear in the metadata. I don't think that we need to recode these fields, since they seem fairly well defined. The format could simply repeat the value as a label: `"APPLICATION ACCEPTED" = "APPLICATION ACCEPTED" "APPLICATION APPROVED" = "APPLICATION APPROVED"` etc.

Reply:
@ptatian can you remind me what the proper way is to save formats for raw data? Should the proc format be in its own program or in the read_building_permits macro?

Reply:
@ptatian Never mind, I found an example and used that.